resourcename/hasparent_test.go

```diff
@@ -83,6 +83,34 @@ func TestHasParent(t *testing.T) {
 			parent:   "//freight-example.einride.tech/shippers/-",
 			expected: false,
 		},
+
+		{
+			test:     "revisioned child",
+			name:     "shippers/1/sites/1@beef",
+			parent:   "shippers/1/sites/1",
+			expected: true,
+		},
+
+		{
+			test:     "revisioned child",
+			name:     "shippers/1/sites/1@beef",
+			parent:   "shippers/1/sites/1",
+			expected: true,
+		},
+
+		{
+			test:     "revisioned child",
+			name:     "shippers/1/sites/1@beef",
+			parent:   "shippers/1/sites/1@dead",
+			expected: false,
+		},
+
+		{
+			test:     "identical revisioned child",
+			name:     "shippers/1/sites/1@beef",
+			parent:   "shippers/1/sites/1@beef",
+			expected: false,
+		},
```

Comment:
Missing tests for resource names where the revision is not in the last segment of the name. For example:
* `parent: "datasets/1@beef", name: "datasets/1@beef/tables/1", expected: true`
* `parent: "datasets/1@beef", name: "datasets/1/tables/1", expected: false`
Not 100% about this one yet.
* `parent: "datasets/1@beef", name: "datasets/1@dead/tables/1", expected: false`

Reply:
Perfect! Adding these

Reply:
I agree with the 3rd one

Reply:
Updated